tests/unit/test_local_iterators.py

```diff
@@ -0,0 +1,44 @@
+import numpy as np
+import random
+from autofaiss.datasets.readers.local_iterators import read_embeddings_local
+import shutil
+import os
+
+
+def build_test_collection(min_size=2, max_size=10000, dim=512, nb_files=5):
+    tmp_dir = "/tmp/autofaiss_test"
+    if os.path.exists(tmp_dir):
+        shutil.rmtree(tmp_dir)
+    os.mkdir(tmp_dir)
+    sizes = [random.randint(min_size, max_size) for _ in range(nb_files)]
+    dim = dim
+    for i, size in enumerate(sizes):
+        arr = np.random.rand(size, dim).astype("float32")
```

Comment:
We might want to test the code for float16 and float64 too

Reply:
yeah it would be useful to test that